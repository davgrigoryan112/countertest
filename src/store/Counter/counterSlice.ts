import { CounterActionType, CounterState, CounterActions } from "./types";
import { createSlice } from '@reduxjs/toolkit'

const initialState: CounterState = {
    count: 0
};

export const counterSlice = createSlice({
    name: 'count',
    initialState,
    reducers: {
        increment: (state) => {
            state.count += 1
        },
        decrement: (state) => {
            state.count -= 1
        },
    },
})

// Action creators are generated for each case reducer function
export const { increment, decrement } = counterSlice.actions

export default counterSlice.reducer




//
// const count = (
//     state: CounterState = initialState,
//     action: Action | CounterActions
// ): CounterState => {
//     switch (action.type) {
//         case CounterActionType.INCREMENT:
//             return { ...state, count: state.count + 1 };
//         case CounterActionType.DECREMENT:
//             return { ...state, count: state.count - 1 };
//         default:
//             return state;
//     }
// };
//
// export default count;


