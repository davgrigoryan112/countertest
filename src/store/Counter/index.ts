export * from './counterSlice'
export * from './types'
