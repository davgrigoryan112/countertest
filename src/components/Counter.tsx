import * as React from "react";
import { useDispatch, useSelector } from "react-redux";
import {FC} from "react";

import { Button, CountContainer, MainContainer } from "./styles/styled";
import {
    incrementCounter,
    decrementCounter
} from "../store/Counter/";
import { RootState } from "../store";

const Counter: FC = () => {

    const count = useSelector( (state: RootState ) => state.count)
    const dispatch = useDispatch()

    const onIncrement = () => {
        dispatch(incrementCounter())
    }
    const onDecrement = () => {
        dispatch(decrementCounter())
    }

    return (
        <MainContainer>
            <Button id='increment' onClick={onIncrement}><span>+1</span></Button>
            <CountContainer id='count'>{count.count}</CountContainer>
            <Button id='decrement' onClick={onDecrement}><span>-1</span></Button>
        </MainContainer>
    );
};


export default Counter;
