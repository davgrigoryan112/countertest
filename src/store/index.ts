import { combineReducers, compose } from "redux";
import { configureStore } from '@reduxjs/toolkit'
import count from "./Counter/reducer";
import { CounterState } from "./Counter/types";

const reducers = combineReducers({
    count
});


export interface RootState {
    count: CounterState;
}

const composeEnhancers =
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = configureStore({ reducer: reducers, enhancers: composeEnhancers })

export default store;
