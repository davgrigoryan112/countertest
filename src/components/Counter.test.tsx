import {fireEvent, render} from "@testing-library/react";
import Counter from "./Counter";
import {Provider} from "react-redux";
import store from "../store";



describe('CounterTest',() => {

    test('should increment by 1 when clicking increment button',()=>{
        const {baseElement} = render(
            <Provider store={store}>
                <Counter />
            </Provider>
        );
        const incrementButton = baseElement.querySelector('#increment')
        const count = baseElement.querySelector('#count')
        expect(count?.innerHTML).toEqual('0')


        fireEvent.click(incrementButton as Element)
        fireEvent.click(incrementButton as Element)
        fireEvent.click(incrementButton as Element)

        expect(count?.innerHTML).toEqual('3')
    })
    test('should decrement by 1 when clicking decrement button',()=>{
        const {baseElement} = render(
            <Provider store={store}>
                <Counter />
            </Provider>
        );
        const decrementButton = baseElement.querySelector('#decrement')
        const count = baseElement.querySelector('#count')
        expect(count?.innerHTML).toEqual('3')

        fireEvent.click(decrementButton as Element)
        fireEvent.click(decrementButton as Element)

        expect(count?.innerHTML).toEqual('1')

    })
})

