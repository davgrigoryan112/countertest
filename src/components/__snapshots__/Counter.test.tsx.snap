// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ byJfrz"
    >
      <button
        class="sc-gsnTZi jrQkrj"
        id="increment"
      >
        <span>
          +1
        </span>
      </button>
      <div
        class="sc-dkzDqf lhZAug"
        id="count"
      >
        0
      </div>
      <button
        class="sc-gsnTZi jrQkrj"
        id="decrement"
      >
        <span>
          -1
        </span>
      </button>
    </div>
  </div>
</body>
`;
